stjude/proteinpaint
# Matrix sorter UI

## Description

NOTE: The new sorter UI requires a dataset matrix settings option to be set to true (see manual testing B below). We can remove this feature/settings flag once the feature is approved.

To test:
A. Automated (does not require setting a dataset settings option)
- http://localhost:3000/testrun.html?dir=plots&name=matrix.sort.unit
- http://localhost:3000/testrun.html?dir=plots&name=matrix.sorterUi.unit
- http://localhost:3000/testrun.html?dir=mass&name=matrix.integration

B. Manual:
- in `sjpp/gdc/active/dataset/gdc.hg38.ts`, add `useSorterUi: true`  in `termdb.matrix.settings`
- you should see the new UI in http://localhost:3000/example.gdc.matrix.html?maxGenes=3&cohort=Gliomas
- you can open a table row in the sorter UI and then enable, check "Use Order", or drag entries to change the order of  tiebreakers/values.

![Screenshot 2024-04-24 at 4 43 37 PM](https://github.com/stjude/proteinpaint/assets/411031/5e21cc13-d53e-42ad-9a65-129ecd25d037)

 

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
